### ISSUE TITLE
2d Spiral primative

### ISSUE THREAD

**Issue by fragmuffin** [dcowden/cadquery#196]
In the interest of #195, we also need a method of making a spiral primative

something like:

```python
import cadquery
from Helpers import show
part = cadquery.Workplane("XY") \
    .spiral(radius=1, growth=2, radius=108).close() \
    .extrude(3)
show(part)
```

looking into https://freecadweb.org/wiki/Part_Spiral/en now

**Comment by fragmuffin** [comment]
FreeCAD's spiral primitive implementation:
[`App::DocumentObjectExecReturn *Spiral::execute(void)`](https://github.com/FreeCAD/FreeCAD/blob/949c86d364f05eecb5e7c279d6b2bc20e76e65ec/src/Mod/Part/App/PrimitiveFeature.cpp#L907)

It makes a wire along a conical service, then projects it onto a plane.
And there doesn't seem to be an existing python hook for it.

**Comment by dcowden** [comment]
This has been closed in favor of the CQ 2.0 implemention at https://github.com/cadquery/cadquery
